PolyFill.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace GrayBMP;
+
+#region class PolyFill -----------------------------------------------------------------------------
+class PolyFill {
+   #region Methods ---------------------------------------------------
+   public void AddLine (int x1, int y1, int x2, int y2) => mLines.Add ((x1, y1, x2, y2));
+
+   public void Fill (GrayBMP bmp) {
+      for (int i = 0; i < bmp.Height; i++) {
+         var xInters = new List<int> ();
+         foreach (var line in mLines)
+            if (GetXIntersection (line.X1, line.Y1, line.X2, line.Y2, i + 0.5, out double res)) xInters.Add ((int)res);
+         xInters = xInters.Order ().ToList ();

```

Comment:
You are creating **2** list objects on each iteration. Try to rewrite so you create only one List<int> for the entire Fill method, and keep reusing it. It is much faster to Clear() and reuse an existing list, than to make a new one.